src/main/java/pl/edu/zut/mad/schedule/ScheduleFragment.kt

```diff
@@ -0,0 +1,93 @@
+package pl.edu.zut.mad.schedule
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.support.design.widget.Snackbar
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.ognev.kotlin.agendacalendarview.builder.CalendarContentManager
+import com.ognev.kotlin.agendacalendarview.models.CalendarEvent
+import kotlinx.android.synthetic.main.fragment_schedule.*
+import org.joda.time.LocalDate
+import pl.edu.zut.mad.schedule.login.LoginActivity
+import pl.edu.zut.mad.schedule.module.ScheduleModule
+import pl.edu.zut.mad.schedule.util.app
+import java.util.*
+import javax.inject.Inject
+
+
+class ScheduleFragment : Fragment(), ScheduleMvp.View {
+
+    companion object {
+        private const val REQUEST_CODE = 123
+    }
+
+    var dateListener: DateListener? = null
+    @Inject internal lateinit var presenter: ScheduleMvp.Presenter
+
+    private val calendarContentManager: CalendarContentManager by lazy {
+        CalendarContentManager(CalendarController(dateListener), scheduleCalendarView, LessonsAdapter(activity))
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
+            inflater.inflate(R.layout.fragment_schedule, container, false)
+
+    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        init()
+    }
+
+    private fun init() {
+        initInjections()
+        presenter.loadData()
```

Comment:
I would prefer to have in a presenter more dumb method names. Here something like viewIsCreated or, generic names like attach(), visible(), invisible(), detach(). Let presenter decide when to load data.